pandas/tests/indexing/interval/test_interval_new.py

```diff
@@ -143,9 +144,27 @@ def test_loc_with_slices(self):
         tm.assert_series_equal(expected, s[:2.5])
         tm.assert_series_equal(expected, s[0.1:2.5])
 
-        # slice of scalar with step != 1
-        with pytest.raises(ValueError):
-            s[0:4:2]
+    def test_slice_step_ne1(self):
+        # GH#31658 slice of scalar with step != 1
+        s = self.s
+        expected = s.iloc[0:4:2]
+
+        result = s[0:4:2]
+        tm.assert_series_equal(result, expected)
+
+        result2 = s[0:4][::2]
+        tm.assert_series_equal(result2, expected)
+
+    def test_slice_interval_step(self):
+        # GH#31658 allows for integer step!=1, not Interval step
+        s = self.s
+        msg = (
+            "cannot do slice indexing on "
+            "<class 'pandas.core.indexes.interval.IntervalIndex'> with "
+            "these indexers .* of <class 'pandas._libs.interval.Interval'>"
```

Comment:
The current error message I get is more comprehensible I think (with `s` from the tests here):

```
In [41]: s[Interval(3, 4):Interval(4, 5):Interval(1, 2)]                                                                                                                                                           
...
ValueError: cannot support not-default step in a slice

In [42]: s[1:2:Interval(1, 2)]                                                                                                                                                           
...
ValueError: cannot support not-default step in a slice
```